src/vs/vscode.d.ts

```diff
@@ -4849,16 +4889,19 @@ declare module 'vscode' {
 		export const onDidSaveTextDocument: Event<TextDocument>;
 
 		/**
-		 * Get a configuration object.
+		 * Get a workspace configuration object.
 		 *
 		 * When a section-identifier is provided only that part of the configuration
 		 * is returned. Dots in the section-identifier are interpreted as child-access,
 		 * like `{ myExt: { setting: { doIt: true }}}` and `getConfiguration('myExt.setting').get('doIt') === true`.
 		 *
+		 * When a resource is provided, configuration scoped to that resource is returned.
+		 *
 		 * @param section A dot-separated identifier.
-		 * @return The full workspace configuration or a subset.
+		 * @param resource A resource for which the configuration is asked for
+		 * @return The full configuration or a subset.
 		 */
-		export function getConfiguration(section?: string): WorkspaceConfiguration;
+		export function getConfiguration(section?: string, resource?: Uri): WorkspaceConfiguration;
```

Comment:
I would like to call `resource` because extension author can give any resource to compute the configuration for. Don't want to restrict to pass only `workspaceFolder`. 

I use `workspaceFolder` mainly for the configuration target and for the `workspaceFolderValue` in `inspect` because it tells that a configuration value is being read/written from workspace folder settings.  Not sure the terminology `ConfigurationTarget.Resource` & `resourceValue` explains the above intent?

How about naming `resource` in `getConfiguration(section, resource)` to `workspaceFolderResource` ?